core/src/main/java/tech/tablesaw/joining/DataFrameJoiner.java

```diff
@@ -198,31 +199,60 @@ private Table joinInternal(
 
     Selection table1DoneRows = Selection.with();
     Selection table2DoneRows = Selection.with();
-    for (Row row : table1) {
-      int ri = row.getRowNumber();
-      if (table1DoneRows.contains(ri)) {
-        // Already processed a selection of table1 that contained this row.
-        continue;
-      }
-
-      Selection table1Rows = createMultiColSelection(table1, ri, table1Indexes, table1.rowCount());
-      Selection table2Rows = createMultiColSelection(table1, ri, table2Indexes, table2.rowCount());
+    // 2021/04/24 @Daniel Modified: If size of table1 is larger than table2,
```

Comment:
I would remove "2021/04/24 @Daniel Modified:" from the comment. That's what Git history is for

Reply:
Thanks for your review: ) I deleted the comment and push another commit. Please check for the latest version.